### PR TITLE
Edited step sizes and frequencies

### PR DESCRIPTION
**Pull request by kbhavadharani2**

### CODE REVIEW

File: Dataset_generation.m [hunk 1]
```diff
@@ -17,8 +17,8 @@
 for i = 1:snr_len
     x = "Normal";
     t = [0: ts :0.2-ts];                   % 640 sample points per disturbance
-    for f = 49.5:0.025:50.5                % 1000 different sine waves  (Runs 40 times)
-        for vm=0.97:0.0401606:1.03         % (Runs 25 times)
+    for f = 49.5:0.0245:50.5                % 1000 different sine waves  (Runs 40 times)
+        for vm=0.95:0.00396:1.05         % (Runs 25 times)
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you please check the step size once again.

File: Dataset_generation.m [hunk 2]
```diff
@@ -390,34 +401,37 @@
...
-                    t2=t1+0.06;               %3 cycles
-                    y = (1-alpha*((heaviside(t-t1)-heaviside(t-t2)))).*(alpha1* sin(2*pi*f*t)+ alpha3*sin(3*2*pi*f*t)+ alpha5*sin(5*2*pi*f*t));
-                    z= vertcat(z,y);
-                    cl=vertcat(cl,x);
-                    %}
+
+for alpha=0.1:0.2051282051:0.9                         % Runs 4 times  
+    for t1=0.04:0.0036734694:0.058                    % Runs 5 times
+        for f=49.9:0.1:50                        % Runs 2 times 
+            for alpha3=0.05:0.025:0.15               % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you please check the step size.

File: Dataset_generation.m [hunk 3]
```diff
@@ -441,30 +455,32 @@
...
-                    t2=t1+0.06;               %3 cycles
-                    y = (1+alpha*((heaviside(t-t1)-heaviside(t-t2)))).*(alpha1* sin(2*pi*f*t)+ alpha3*sin(3*2*pi*f*t)+ alpha5*sin(5*2*pi*f*t));
-                    z= vertcat(z,y);
-                    cl=vertcat(cl,x);
-                    %}
+
+for alpha=0.1:0.1794871795:0.8                         % Runs 4 times  
+    for t1=0.04:0.0036734694:0.058                    % Runs 5 times
+        for f=49.9:0.1:50                        % Runs 2 times
+            for alpha3=0.05:0.025:0.15               % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 4]
```diff
@@ -489,30 +505,32 @@
...
-                    t2=t1+0.06;               %3 cycles
-                    y = (1-alpha*((heaviside(t-t1)-heaviside(t-t2)))).*(alpha1* sin(2*pi*f*t)+ alpha3*sin(3*2*pi*f*t)+ alpha5*sin(5*2*pi*f*t));
-                    z= vertcat(z,y);
-                    cl=vertcat(cl,x);
-                    %}
+
+for alpha=0.9:0.0256410256:1                           % Runs 4 times  
+    for t1=0.04:0.0036734694:0.058                    % Runs 5 times
+        for f=49.9:0.1:50                        % Runs 2 times
+            for alpha3=0.05:0.025:0.15               % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 5]
```diff
@@ -537,16 +555,17 @@
...
+for alpha_flicker=0.08:0.0307692308:0.2                 % Runs 4 times
     for beta=5:3.703703704:20                          % Runs 5 times
-        for alpha3=0.05:0.025:0.15               % Runs 5 times
-            for alpha5=0.05:0.02469135802:0.15           % Runs 5 times 
-                alpha1 = sqrt(1 - alpha3^2 - alpha5^2);
-                y = (1+alpha_flicker*sin(beta*2*pi*f*t)).*(alpha1* sin(2*pi*f*t)+ alpha3*sin(3*2*pi*f*t)+ alpha5*sin(5*2*pi*f*t));
-                z= vertcat(z,y);
-                cl=vertcat(cl,x); 
+        for f=49.9:0.1:50                        % Runs 2 times
+            for alpha3=0.05:0.025:0.15               % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 6]
```diff
@@ -571,27 +590,29 @@
...
-                y = (1+alpha_flicker*sin(beta*2*pi*f*t)).*sin(2*pi*f*t).*(1-alpha*((heaviside(t-t1)-heaviside(t-t2))));
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                %}
+
+for alpha=0.1:0.2051282051:0.9                          % Runs 4 times  
+    for t1=0.04:0.0036734694:0.058                      % Runs 5 times
+        for f=49.9:0.1:50                        % Runs 2 times
+            for alpha_flicker=0.08:0.0244897959:0.2          % Runs 5 times
+                for beta=5:3.703703704:20                      % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 7]
```diff
@@ -616,27 +637,29 @@
...
 
-for alpha=0.1:0.077:0.8                          % Runs 10 times  
-    for t1=0.04:0.0044444:0.058                      % Runs 5 times
-        for alpha_flicker=0.08:0.02469135802:0.2          % Runs 5 times
+
+for alpha=0.1:0.1794871795:0.8                          % Runs 4 times  
+    for t1=0.04:0.0036734694:0.058                      % Runs 5 times
+        for f=49.9:0.1:50                        % Runs 2 times
+        for alpha_flicker=0.08:0.0244897959:0.2          % Runs 5 times
             for beta=5:3.703703704:20                      % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 8]
```diff
@@ -664,25 +687,27 @@
...
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                
-                t4=t3+0.04;          %2 cycles
-                y= (1-alpha*((heaviside(t-t1)-heaviside(t-t2)))) *(alpha*(heaviside(t-t3)-heaviside(t-t4)).*exp(t3-t/tau).*sin(2*pi*F_t*t));
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                %}
+for alpha=0.1:0.2051282051:0.9                         % Runs 4 times
+    for F_t=300:1160.49382:5000                    % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 9]
```diff
@@ -664,25 +687,27 @@
...
+        for f=49.9:0.1:50                        % Runs 2 times
+            for t3=0.04:0.01:0.08                   % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 10]
```diff
@@ -664,25 +687,27 @@
...
+                for tau=0.008:0.007901234568:0.040          % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 11]
```diff
@@ -708,27 +733,29 @@
 
 x = "Oscillatory Transient + Swell ";                            
 t = [0: ts :0.2-ts];                   % 640 sample points per disturbance
-f = 50;
 
-for alpha=0.1:0.0773480663:0.9                         % Runs 10 times
+
+for alpha=0.1:0.2051282051:0.9                         % Runs 4 times
     for F_t=300:1160.493827:5000                    % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 12]
```diff
@@ -708,27 +733,29 @@
...
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                
-                t4=t3+0.04;          %2 cycles
-                y= (1+alpha*((heaviside(t-t1)-heaviside(t-t2)))) + sin(2*pi*f*t)+ alpha*(heaviside(t-t3)-heaviside(t-t4)).*exp(t3-t/tau).*sin(2*pi*F_t*t);
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                %}
+        for f=49.9:0.1:50                        % Runs 2 times
+            for t3=0.04:0.01:0.08                   % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 13]
```diff
@@ -708,27 +733,29 @@
...
+                for tau=0.008:0.007901234568:0.040          % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 14]
```diff
@@ -754,27 +781,29 @@
 
 x = "Oscillatory Transient + Interruptions";                            
 t = [0: ts :0.2-ts];                   % 640 sample points per disturbance
-f = 50;
 
-for alpha=0.1:0.0773480663:0.9                         % Runs 10 times
+
+for alpha=0.1:0.2051282051:0.9                         % Runs 4 times
     for F_t=300:1160.493827:5000                    % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

File: Dataset_generation.m [hunk 15]
```diff
@@ -754,27 +781,29 @@
...
-                cl=vertcat(cl,x);
-                
-                t4=t3+0.04;          %2 cycles
-                y= (1-alpha*((heaviside(t-t1)-heaviside(t-t2)))) + sin(2*pi*f*t)+ alpha*(heaviside(t-t3)-heaviside(t-t4)).*exp(t3-t/tau).*sin(2*pi*F_t*t);
-                z= vertcat(z,y);
-                cl=vertcat(cl,x);
-                %}
+        for f=49.9:0.1:50                        % Runs 2 times
+            for t3=0.04:0.01:0.08                   % Runs 5 times
+                for tau=0.008:0.007901234568:0.040          % Runs 5 times
```
Comment (Surendra-Srinivas) [inline Dataset_generation.m]: Can you check the step size.

**Review by Surendra-Srinivas — CHANGES_REQUESTED** [review]
Hi can you check the comments I mentioned at some lines. Acknowledge once done for merging.

File: Dataset_generation.m (hunk 15 above)
Comment (kbhavadharani2) [reply Dataset_generation.m]: for t3 = 0.04, 0.05, 0.06, 0.07, 0.08
for tau ill chang (0.0065306122)

File: Dataset_generation.m (hunk 1 above)
Comment (kbhavadharani2) [reply Dataset_generation.m]: done

**Comment by Surendra-Srinivas** [comment]
@kbhavadharani2 
There were 40 Merge Conflicts, it was hard to resolve. Anyways I pulled it all. 
And I found that there are no frequency loops for last two classes, Can you check that out ?

**Pull request merged**